internal/cli/version.go

```diff
@@ -0,0 +1,141 @@
+`
+
+const UpdateVersionMsgFormat string = `
+  We need to update your New Relic CLI version to continue.
+
+    Installed version: %s
+    Latest version:    %s
+
+  To update your CLI and continue this installation, run this command:
+
+    %s
+`
+
+// TODO: Refactor this into a singleton object that can be shared across packages
+
+// IsLatestVersion returns true if the provided version string matches
+// the current installed version.
+func IsLatestVersion(ctx context.Context, latestVersion string) (bool, error) {
+	installedVersion, err := semver.NewVersion(Version)
+	if err != nil {
+		return false, fmt.Errorf("error parsing current CLI version %s: %s", Version, err.Error())
+	}
+
+	lv, err := semver.NewVersion(latestVersion)
+
+	log.Printf("\n latest version prerelease:     %+v \n", lv.Prerelease())
+	log.Printf("\n installed version prerelease:  %+v \n", installedVersion.Prerelease())
+
+	if err != nil {
+		return false, fmt.Errorf("error parsing version to check %s: %s", latestVersion, err.Error())
+	}
+
+	if installedVersion.LessThan(lv) {
+		return false, nil
+	}
+
+	return installedVersion.Equal(lv), nil
+}
+
+func GetLatestReleaseVersion(ctx context.Context) (string, error) {
+	latestRelease, err := FetchLatestRelease(ctx)
+	if err != nil {
+		return "", fmt.Errorf("error fetching latest release: %s", err.Error())
+	}
+
+	return latestRelease.TagName, nil
+}
+
+func FetchLatestRelease(ctx context.Context) (*GitHubRepositoryTagResponse, error) {
+	client := utils.NewHTTPClient()
+
+	respBytes, err := client.Get(ctx, NewRelicCLILatestReleaseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	repoTag := GitHubRepositoryTagResponse{}
+	err = json.Unmarshal(respBytes, &repoTag)
+	if err != nil {
+		return nil, err
+	}
+
+	log.WithFields(log.Fields{
+		"tag": repoTag.TagName,
+	}).Debug("fetch tag success")
+
+	return &repoTag, nil
+}
+
+// IsDevEnvironment returns true when the installed CLI version
+// is either in a prerelease state or in a dirty state. The version
+// string is generated at compile time using git. The prerelease string
+// is appended to the primary semver version string.
+//
+// Examples of prerelease version strings
+//
+//  v0.32.1-10-gbe63a24
+//  v0.32.1-10-gbe63a24-dirty
+//
+// In this example version string, "10" represents the number of
+// commits since the 0.32.1 tag was created. The "gbe63a24" is the
+// previous commit's abbreviated sha. The "dirty" part means that
+// git was in a dirty state at compile time, meaning an updated file
+// was saved, but not yet committed.
+func IsDevEnvironment() bool {
+	v, err := semver.NewVersion(Version)
+	if err != nil {
+		return true
+	}
+
+	prereleaseString := v.Prerelease()
+	hasPrereleaseString := prereleaseString != ""
+
+	if hasPrereleaseString && strings.Contains(prereleaseString, "dirty") {
+		return true
+	}
+
+	if hasPrereleaseString && !strings.Contains(prereleaseString, "alpha") && !strings.Contains(prereleaseString, "beta") {
```

Comment:
When we're in a dev environment, we're "always behind" the latest version, however, some prerelease strings might contain `alpha` or `beta` (e.g. `v1.0.0-beta` etc). Technically we've never release this kind of tag, but I wanted to have defensive code in place just in case. 

This scenario is fairly unlikely for the foreseeable future, so we can remove it if we don't want it. 

e.g.